build/terraform/gke/module.tf

```diff
@@ -63,6 +63,11 @@ variable "ping_service_type" {
   default = "LoadBalancer"
 }
 
+variable "location" {
+  default     = ""
+  description = "The GCP location to create the cluster in"
+}
+
```

Comment:
Should we just deprecate `zone`? IMO yes, but would like input from @roberthbailey. It would look like:
* on new line 67, change the default to `"us-west1-c"`
* on new line 72, change the default to `""`
* on new line 73, let's say `"The GCP zone to create the cluster in (deprecated, use 'location')"`

It looks like there is no formal deprecation procedure (https://github.com/hashicorp/terraform/issues/18381) though, so I think to actually enforce the deprecation we'd have to add output to the module if `zone` were set.

WDYT?

Reply:
I think the plan is to 
> make sure we keep things backwards compatible for at least 1 release and give folks a warning in the release notes that the variable name is changing.

So adding a deprecation note does follow that. But does this mean that `zone` would take precedence over `location` for the 1 release before being removed?

Reply:
`GCP_CLUSTER_ZONE` and `GCP_CLUSTER_LOCATION` are both set, but I think the change you just made to `gcloud-terraform-install` sets only location, right?

The logic is a little fussy, but I think we should check:

* if `location` is set but not `zone`, use `location` (good path)
* if `zone` is set but not `location`, use `zone` and warn (deprecated path)
* if `location` and `zone` are set but equal, use either and warn.
* if *both* are set, and not equal, throw an error

I'm not exactly sure how possible that is in .tf, but it seems like the right approach. If it's too much effort, though, holler.

Reply:
I will add back `zone` to `gcloud-terraform-install` so that both variables will be available. As for adding warnings and errors, not sure if that is possible within the terraform files?

Reply:
I don't think you need to add `zone` back in, I was pointing out that the Makefile is only sending the one - so there's no conflict between them.

Reply:
Okay so I made it so that within the tf file, zone will be empty unless explicitly set. And it will only use zone if the variable is empty.

Reply:
I didn't see this comment thread earlier but I think where you ended up looks good - we use location and we can add to the release notes that folks should switch their configs to use location but we continue to support zone for N releases (maybe 1, maybe a few) to give people time to migrate their configs. 

We don't have a formal policy about compatibility of terraform configs (or helm for that matter) but we do try to do our best not to have knife edge changes that would break users upgrading across a single release.